subprojects/resources-gcs/src/integTest/groovy/org/gradle/integtests/resource/gcs/GcsClientIntegrationTest.groovy

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resource.gcs
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential
+import com.google.api.client.http.HttpRequest
+import com.google.api.client.http.HttpRequestInitializer
+import com.google.api.client.http.javanet.NetHttpTransport
+import com.google.api.client.json.jackson2.JacksonFactory
+import com.google.api.services.storage.Storage
+import com.google.api.services.storage.model.StorageObject
+import org.apache.commons.io.IOUtils
+import org.gradle.integtests.resource.gcs.fixtures.GcsServer
+import org.gradle.internal.resource.transport.gcp.gcs.GcsClient
+import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
+import org.junit.Rule
+import spock.lang.Ignore
+import spock.lang.Specification
+
+class GcsClientIntegrationTest extends Specification {
+
+    public static final String FILE_NAME = "mavenTest.txt"
+    final String bucketName = 'org.gradle.artifacts'
+
+    @Rule
+    final TestNameTestDirectoryProvider temporaryFolder = new TestNameTestDirectoryProvider()
+
+    @Rule
+    public final GcsServer server = new GcsServer(temporaryFolder)
```

Comment:
Junit's rule need to be public for the runner to access it reflectively: http://junit.org/junit4/javadoc/4.12/org/junit/Rule.html